src/category_theory/monoidal/internal.lean

```diff
@@ -39,6 +39,16 @@ attribute [simp, reassoc] Mon_.one_mul Mon_.mul_one Mon_.mul_assoc
 
 namespace Mon_
 
+theorem hext (M N : Mon_ C) : M.X = N.X → M.one == N.one → M.mul == N.mul → M = N :=
```

Comment:
Is this still used? Perhaps we could remove it as "evil" if not.

Reply:
Yeah, I use this and the Monad analogue in `of_to_mon_end_obj` and `to_of_mon_end_obj`. I think these theorems could be useful. What do you think?

Reply:
I just pushed a small change to these two theorems, which I think was necessary for them to be tagged as `simp`.

Reply:
Sounds good!

Reply:
Actually, now I think that `of_to_mon_end_obj` and `to_of_mon_end_obj` may not be fit for purpose. You don't use them in this PR, and as they are not definitional equalities it's going to be hard to use them --- you can't use them with `dsimp`, and `rw` and `simp` will fail to apply these equalities when you're looking at morphisms, because they'll break the dependent types.

So perhaps delete them, and this `hext`?

Reply:
@semorrison Fair enough. I'll go ahead and delete these theorems, and the `hext` theorems for `Mon_` and `Monoid` before merging.